### PR TITLE
Redis auth support

### PR DESCRIPTION
**Pull request by furiel**
For issue:
https://github.com/balabit/syslog-ng/issues/1320

```
Example config:
destination d_redis {
    redis(
	host("127.0.0.1")
	port(6379)
        auth("Pa55W0rD") # <---
	command("HINCRBY", "hosts", "$HOST", "1")
    );
};
```

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build? (admin please type: ok to test)

**Comment by pzoleex** [comment]
@kira-syslogng ok to test

File: modules/redis/redis.c [hunk 1]
```diff
@@ -135,6 +145,32 @@ redis_dd_format_persist_name(const LogPipe *s)
 }
 
 static gboolean
+sendRedisCommand(RedisDriver *self, const char *format, ...)
```
Comment (kvch) [inline modules/redis/redis.c]: Please don't use camel case in names.

File: modules/redis/redis.c [hunk 2]
```diff
@@ -135,6 +145,32 @@ redis_dd_format_persist_name(const LogPipe *s)
...
+  va_end(ap);
+
+  gboolean retval = reply && (reply->type != REDIS_REPLY_ERROR);
+  if (reply)
+    freeReplyObject(reply);
+  return retval;
+}
+
+static gboolean
+test_connection_to_redis(RedisDriver *self)
```
Comment (kvch) [inline modules/redis/redis.c]: Could you rename `test_*` functions to `check_*`? It is confusing to see a function starting with `test_` outside the test code.

File: modules/redis/redis.c [hunk 3]
```diff
@@ -76,6 +77,15 @@ redis_dd_set_port(LogDriver *d, gint port)
 }
 
 void
+redis_dd_set_auth(LogDriver *d, const gchar *auth)
+{
+  RedisDriver *self = (RedisDriver *)d;
+
+  g_free(self->auth);
+  self->auth = g_strdup(auth);
```
Comment (kvch) [inline modules/redis/redis.c]: Please, free `self->auth` in `redis_dd_free`.

**Comment by lbudai** [comment]
@furiel : could you write a more detailed commit message?
(url to the redis doc how this authentication works, and an example syslog-ng configuration snippet would be very useful )

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/redis/redis.c (hunk 3 above)
Comment (furiel) [inline modules/redis/redis.c]: fixed

Comment (kvch) [inline modules/redis/redis.c]: :+1:

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/cfg-grammar.y [hunk 4]
```diff
@@ -330,6 +330,9 @@ extern struct _StatsOptions *last_stats_options;
 
 %token KW_RETRIES                     10511
 
+/* authentication */
+%token KW_AUTH                        10600
```
Comment (MrAnno) [inline lib/cfg-grammar.y]: You can declare this token in `redis-grammar.ym` if you don't want to use it in multiple modules.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by lbudai** [comment]
:+1:

**Pull request merged**